homeassistant/components/geonetnz_quakes/sensor.py

```diff
@@ -0,0 +1,130 @@
+"""Feed Entity Manager Sensor support for GeoNet NZ Quakes Feeds."""
+import logging
+from typing import Optional
+
+from homeassistant.components.geonetnz_quakes import DOMAIN, FEED
+from homeassistant.components.geonetnz_quakes.geo_location import SIGNAL_STATUS
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_STATUS = "status"
+ATTR_LAST_UPDATE = "last_update"
+ATTR_LAST_UPDATE_SUCCESSFUL = "last_update_successful"
+ATTR_LAST_TIMESTAMP = "last_timestamp"
+ATTR_CREATED = "created"
+ATTR_UPDATED = "updated"
+ATTR_REMOVED = "removed"
+
+DEFAULT_ICON = "mdi:pulse"
+DEFAULT_UNIT_OF_MEASUREMENT = "quakes"
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the GeoNet NZ Quakes Feed platform."""
+    sensor = GeonetnzQuakesSensor(entry.entry_id, entry.title)
+    async_add_entities([sensor])
+
+
+class GeonetnzQuakesSensor(Entity):
+    """This is a status sensor for the GeoNet NZ Quakes integration."""
+
+    def __init__(self, config_entry_id, config_title):
+        """Initialize entity."""
+        self._config_entry_id = config_entry_id
+        self._config_title = config_title
+        self._status = None
+        self._last_update = None
+        self._last_update_successful = None
+        self._last_timestamp = None
+        self._total = None
+        self._created = None
+        self._updated = None
+        self._removed = None
+        self._remove_signal_status = None
+
+    async def async_added_to_hass(self):
+        """Call when entity is added to hass."""
+        self._remove_signal_status = async_dispatcher_connect(
+            self.hass,
+            SIGNAL_STATUS.format(self._config_entry_id),
+            self._update_status_callback,
+        )
+        _LOGGER.debug("Waiting for updates %s", self._config_entry_id)
+        # First update is manual because of how the feed entity manager is updated.
+        await self.async_update()
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Call when entity will be removed from hass."""
+        if self._remove_signal_status:
+            self._remove_signal_status()
+
+    @callback
+    def _update_status_callback(self):
+        """Call status update method."""
+        _LOGGER.debug("Received status update for %s", self._config_entry_id)
+        self.async_schedule_update_ha_state(True)
+
+    @property
+    def should_poll(self):
+        """No polling needed for GeoNet NZ Quakes status sensor."""
+        return False
+
+    async def async_update(self):
+        """Update this entity from the data held in the feed manager."""
+        _LOGGER.debug("Updating %s", self._config_entry_id)
+        manager = self.hass.data[DOMAIN][FEED][self._config_entry_id]
```

Comment:
Pass the manager when creating the entity and store the manager as an instance attribute.

Reply:
OK. I am still using `hass.data` to store the manager, and extract it in the `async_setup_entry` method.